Tutorial_Android/MyFirstApp/app/src/main/java/com/example/myfirstapp/providers/error/ErrorUtils.java

```diff
@@ -0,0 +1,36 @@
+package com.example.myfirstapp.providers.error;
+
+import com.example.myfirstapp.models.signIn.error.SignInRemoteError;
+import com.example.myfirstapp.providers.RetrofitHelper;
+import com.example.myfirstapp.providers.SignInProvider;
+
+import java.io.IOException;
+import java.lang.annotation.Annotation;
+
+import okhttp3.ResponseBody;
+import retrofit2.Converter;
+import retrofit2.Response;
+
+/**
+ * Created by taqtile on 5/4/17.
+ */
+
+public class ErrorUtils {
```

Comment:
@leonardo-borges, @RodrigoSabino  😱 😱 😱  por favor... evitem nomes como `helper`, `util`. Eles são muito abrangentes e não indicam qual a responsabilidade dela. Aqui vocês podem chamar de `ErrorParser` (pois é isso o que ele faz, né?):

```java
public class ErrorParser {
  // we can simply call as parse instead of parseError since the class name has enough information
  public static SignInRemoteError parse(Response<?> response) {
    ...
  }
}
```